src/example/constrained_poisson/constrained_poisson_multiplierFESpace.jl

```diff
@@ -0,0 +1,130 @@
+module constrained_poisson_API #hide
+println("Running constrained poisson API example...") #hide
+
+# # Constrained Poisson equation (FE)
+# In this example, a Poisson equation with Neumann boundary conditions is solved using a boundary integral constraint.
+#
+# # Theory
+# Consider the following Poisson equation on the unit disk (noted $$\Omega$$ in this example, its boundary is noted $$\Gamma$$):
+# ```math
+#    - \Delta u = f \, \, \forall x \in \Omega
+# ```
+# ```math
+#    \frac{\partial u}{\partial n} = 0 \, \, \forall x \in \Gamma
+# ```
+# Poisson's equation can be written in the form of a minimisation problem:
+# ```math
+#    \min_{u} J(u) = \frac{1}{2} \int_{\Omega} \nabla u . \nabla u \, dV - \int_{\Omega} f u \, dV
+# ```
+# There is no unique solution to this problem (adding a constant to any solution will also be a solution).
+# Uniqueness can be recovered by adding a constraint to the problem. In this example the following constraint is added:
+# ```math
+#    \int_{\Gamma} u \, d \gamma = 2 \pi
+# ```
+# To solve this constrained minimisation problem, the following lagragian is introduced:
+# ```math
+#    \mathcal{L}(u, \lambda_u) = \frac{1}{2} \int_{\Omega} \nabla u . \nabla u \, dV - \int_{\Omega} f u \, dV + \lambda_u ( \int_{\Gamma} u \, d \gamma - 2 \pi)
+# ```
+# where $$\lambda_u$$ is a Lagrange multiplier.
+# The first order optimality conditions translate to the problem: find (u, \lambda_u) such that for all (v, \lambda_v):
+# ```math
+#    \int_{\Omega} \nabla u . \nabla v \, dV + \lambda_u \int_\Gamma v \, d \gamma = \int_\Omega f v \, dV
+# ```
+# ```math
+#    \lamda_v \int_\Gamma u \, d\gamma = 2 \pi \lambda_v
+# ```
+# This problem can be assembled by introducing a MultiplierFESpace and combining it with the usual FESpace using a MultiFESpace. 
+# In this example, the manufactured solution $$u(x,y)=cos(4\pi(x^2 + y^2))$$ is used to test the method.
+
+# import necessary packages
+using Bcube
+using LinearAlgebra
+using SparseArrays
+using WriteVTK
+
+const outputpath = joinpath(@__DIR__, "../../../myout/constrained_poisson/")
+mkpath(outputpath)
+
+# Read 2D mesh
+mesh_path = joinpath(outputpath, "mesh.msh")
+gen_disk_mesh(mesh_path; lc = 3.2e-2)
+mesh = read_msh(mesh_path)
+
+# Choose degree and define function space, trial space and test space
+const degree = 2
+fs = FunctionSpace(:Lagrange, degree)
+U = TrialFESpace(fs, mesh)
+V = TestFESpace(U)
+
+# Define the multiplier trial space and corresponding test space
```

Comment:
maybe specify here that `1` indicates the desired number of multipliers (~ number of scalars)